graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/GraphQLJavaErrorInstrumentation.kt

```diff
@@ -0,0 +1,72 @@
+package com.netflix.graphql.dgs.internal
+
+import com.netflix.graphql.types.errors.ErrorDetail
+import com.netflix.graphql.types.errors.ErrorType
+import com.netflix.graphql.types.errors.TypedGraphQLError
+import graphql.ExecutionResult
+import graphql.GraphQLError
+import graphql.execution.instrumentation.InstrumentationState
+import graphql.execution.instrumentation.SimplePerformantInstrumentation
+import graphql.execution.instrumentation.parameters.InstrumentationExecutionParameters
+import graphql.validation.ValidationError
+import graphql.validation.ValidationErrorType
+import java.util.concurrent.CompletableFuture
+
+class GraphQLJavaErrorInstrumentation : SimplePerformantInstrumentation() {
+    override fun instrumentExecutionResult(executionResult: ExecutionResult, parameters: InstrumentationExecutionParameters?, state: InstrumentationState?): CompletableFuture<ExecutionResult> {
+        if (executionResult.errors.isNotEmpty()) {
+            val newExecutionResult = ExecutionResult.newExecutionResult().from(executionResult)
+            val graphqlErrors: MutableList<GraphQLError> = mutableListOf()
+
+            executionResult.errors.forEach { error ->
+                // put in the classification unless it's already there since graphql-java errors contain this field
+                val extensions = (if (error.extensions != null) error.extensions else emptyMap<String, Any>()).toMutableMap()
+                if (!extensions.containsKey("classification")) {
+                    val errorClassification = error.errorType
+                    extensions["classification"] = errorClassification.toSpecification(error)
+                }
+
+                if (error.errorType == graphql.ErrorType.ValidationError || error.errorType == graphql.ErrorType.InvalidSyntax ||
+                    error.errorType == graphql.ErrorType.NullValueInNonNullableField
+                ) {
+                    val path = if (error is ValidationError) error.queryPath else error.path
+                    val graphqlErrorBuilder = TypedGraphQLError
+                        .newBadRequestBuilder()
+                        .locations(error.locations)
+                        .path(path)
```

Comment:
you are not setting the errorType here, shouldn't it be BAD_REQUEST?

Reply:
it's a badrequest builder, so it will be set there automatically.